### PR TITLE
[MM-38786] Enhancements to handle profiling in a per deployment/pod basis

### PR DESCRIPTION
**Pull request by stylianosrigas**
#### Summary
- Enhancements to handle profiling in a per deployment/pod basis

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-38786

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,8 +13,11 @@ This is the API url that should be used for the profiling attachment uploads. Fo
 - POST_API_URL:
 This is the API url that should be used for the profiling attachment posts. For example `http://community:8065/api/v4/posts`. More information on Mattermost posts API call can be found [here](https://api.mattermost.com/#tag/posts).
 
-- MATTERMOST_PROFILE_TARGETS:
-The k8s service name of the target that the profile will run against to. For example `community`
+- MATTERMOST_DEPLOYMENTS:
+The k8s deployments that profiling will be run against their pods.. For example `community, community-release`
```
Comment (angeloskyratzakos) [inline README.md]:
```suggestion
The k8s deployments that profiling will be run against their pods. For example `community, community-release`
```

**Review by angeloskyratzakos — APPROVED** [review]

File: main.go [hunk 2]
```diff
@@ -26,30 +41,62 @@ type uploadResponse struct {
...
+		}
+		for _, deploymentPod := range deploymentPods.Items {
+			var pod Pod
+			pod.PodName = deploymentPod.GetName()
+			pod.PodIP = deploymentPod.Status.PodIP
+			pods = append(pods, pod)
+		}
+
+		log.Infof("Running profiling for %s", deployment)
+		err = profiling(envVars.ProfilingTime, pods)
```
Comment (spirosoik) [inline main.go]: minor: `if err = profiling(envVars.ProfilingTime, pods); err != nil {}`

File: main.go [hunk 3]
```diff
@@ -26,30 +41,62 @@ type uploadResponse struct {
...
+		if err != nil {
+			log.WithError(err).Error("Failed to run profiling")
+			return
+		}
+		err = uploadPostFiles(pods, *envVars, deployment)
```
Comment (spirosoik) [inline main.go]: minor: `if err = uploadPostFiles(pods, *envVars, deployment); err != nil {}`

File: main.go [hunk 4]
```diff
@@ -71,17 +118,30 @@ func main() {
...
+			return errors.Errorf("Failed to upload file %s", cpuFile)
 		}
 		uploads = append(uploads, cpuResponse.FileInfos[0].ID)
 	}
-
 	log.Info("Posting files")
-	err = postFile(envVars.PostAPIURL, envVars.ChannelID, envVars.Token, uploads, envVars.MattermostProfileTargets)
+	err := postFile(envVars.PostAPIURL, envVars.ChannelID, envVars.Token, deployment, uploads)
+	if err != nil {
+		return errors.Errorf("Failed to post files")
```
Comment (spirosoik) [inline main.go]: better to wrap the error `errors.Wrap(err, "failed to post files")`

File: main.go [hunk 5]
```diff
@@ -71,17 +118,30 @@ func main() {
 		log.Infof("Uploading file %s", cpuFile)
 		cpuResponse, err := uploadFile(envVars.UploadAPIURL, envVars.Token, cpuValues)
 		if err != nil {
-			log.WithError(err).Errorf("Failed to upload file %s", cpuFile)
+			return errors.Errorf("Failed to upload file %s", cpuFile)
```
Comment (spirosoik) [inline main.go]: better to wrap the error `errors.Wrapf(err, "Failed to upload file %s", cpuFile)

**Review by spirosoik — COMMENTED** [review]
minor few comments and changes

File: main.go (hunk 5 above)
Comment (stylianosrigas) [reply main.go]: Thanks fixed

File: main.go (hunk 4 above)
Comment (stylianosrigas) [reply main.go]: Thanks fixed

File: main.go (hunk 3 above)
Comment (stylianosrigas) [reply main.go]: Thanks fixed

File: main.go (hunk 2 above)
Comment (stylianosrigas) [reply main.go]: Thanks fixed

**Review by spirosoik — APPROVED** [review]

**Review by stafot — APPROVED** [review]
LGTM.

**Pull request merged**